odl/space/pspace.py

```diff
@@ -1064,6 +1123,158 @@ def ufuncs(self):
+        >>> x_real = x.real
+        >>> real_space = odl.ProductSpace(odl.rn(3), odl.rn(2))
+        >>> expected_result = real_space.element([[1, 2, 3], [-1, -2]])
+        >>> x_real == expected_result
+        True
+        """
+        real_part = [part.real for part in self.parts]
+        return self.space.real_space.element(real_part)
+
+    @real.setter
+    def real(self, newreal):
+        """Setter for the real part.
+
+        This method is invoked by ``x.real = other``.
+
+        Parameters
+        ----------
+        newreal : array-like or scalar
+            Values to be assigned to the real part of this element.
+
+        Examples
+        --------
+        >>> space = odl.ProductSpace(odl.cn(3), odl.cn(2))
+        >>> x = space.element([[1 + 1j, 2, 3 - 3j], [1 + 1j, 2]])
+        >>> zero = space.real_space.zero()
+        >>> x.real = zero
+        >>> x
+        space.element([[1j, 0j, -3j], [1j, 0]])
+
+        Other array-like types:
+
+        >>> x.real = 1.0
+        >>> x
+        space.element([[1 + 1j, 1, 1 - 3j], [1 + 1j, 1]])
+        >>> x.real = [[2, 3, 4], [5, 6]]
+        >>> x
+        space.element([[2 + 1j, 3, 4 - 3j], [5 + 1j, 6]])
+        """
+        try:
+            iter(newreal)
+        except TypeError:
+            # `newreal` is not iterable, assume it can be assigned to
+            # all indexed parts
+            for part in self.parts:
+                part.real = newreal
+            return
+
+        if self.space.is_power_space:
+            try:
+                # Set same value in all parts
+                for part in self.parts:
+                    part.real = newreal
+            except (ValueError, TypeError):
+                # Iterate over all parts and set them separately
+                for part, new_re in zip(self.parts, newreal):
+                    part.real = new_re
+                pass
+        elif len(newreal) == len(self):
+            for part, new_re in zip(self.parts, newreal):
+                part.real = new_re
+        else:
+            raise ValueError(
+                'dimensions of the new real part does not match the space, '
+                'got element {} to set real part of {}'.format(newreal, self))
+
+    @property
+    def imag(self):
+        """Imaginary part of the element.
+
+        Examples
+        --------
+        >>> space = odl.ProductSpace(odl.cn(3), odl.cn(2))
+        >>> x = space.element([[1 + 1j, 2, 3 - 3j], [-1+2j, -2-3j]])
+        >>> x_imag = x.imag
+        >>> real_space = odl.ProductSpace(odl.rn(3), odl.rn(2))
+        >>> expected_result = real_space.element([[1, 0, -3], [2, -3]])
+        >>> x_imag == expected_result
+        True
+        """
+        imag_part = [part.imag for part in self.parts]
+        return self.space.real_space.element(imag_part)
+
+    @imag.setter
+    def imag(self, newimag):
+        """Setter for the imaginary part.
+
+        This method is invoked by ``x.imag = other``.
```

Comment:
Here, something like `x.imag = 1j` could to a user seem like a natural way of setting the imaginary part to `1`. However, in this case it will be set to `0`, since the imaginary part is a real number and thus set to the real part of `0 + 1j`. Although logical mathematically, it might be confusing before you think about it. Any suggestion on how this can be clarified? Does it need to be clarified?

Reply:
No, it does not need to be clarified since these are somewhat standard from e.g. numpy and other packages and hence easily googleable if anyone wants more info

Reply:
Agreed, we should assume at least this basic amount of knowledge of complex numbers. I don't think `x.imag = 1j` should ever be expected to set the imaginary part to 1.